TPP.ArgsParsing/TypeParsers/ShinyParser.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using TPP.Common;
+
+namespace TPP.ArgsParsing.TypeParsers
+{
+    /// <summary>
+    /// A parser that finds a badge form by name.
+    /// </summary>
+    public class ShinyParser : BaseArgumentParser<bool>
```

Comment:
this clashes with the generic `BoolParser` that has been added on the master branch by now. You need to wrap the enum in a dedicated type to be able to use it for the args parser. You can look at `Pokeyen` for example to see what I mean.